src/uct/sm/cma/cma_iface.c

```diff
@@ -46,6 +47,7 @@ static ucs_status_t uct_cma_iface_query(uct_iface_h tl_iface,
     iface_attr->cap.put.max_zcopy      = SIZE_MAX;
     iface_attr->cap.get.max_zcopy      = SIZE_MAX;
     iface_attr->iface_addr_len         = sizeof(uct_sockaddr_process_t);
+    iface_attr->iface_addr_len         = UCT_SM_IFACE_DEVICE_ADDR_LEN;
```

Comment:
where do you set uct_sm_iface_get_device_address for knem ? for some reason it is still set to ucs_empty_function_return_success